Assets/Scripts/Level/WinConditionChecker.cs

```diff
@@ -0,0 +1,32 @@
+using UnityEngine;
+using UnityAtoms;
+using UniRx;
+using UnityAtoms.FSM;
+
+public class WinConditionChecker : MonoBehaviour {
+
+    [SerializeField]
+    private SceneDataBase dataBase;
+    private int numberOfEnd;
+    [SerializeField]
+    private AtomEvent<bool> goalChangedEvent;
```

Comment:
What do you think of using the standard of naming our event variables "on..."? like `onGoalChanged`